src/data/Dhis2RelationshipTypes.ts

```diff
@@ -294,18 +295,22 @@ async function getConstraintForTypeProgram(
     if (!program) return undefined;
     const { organisationUnits = [], startDate, endDate } = filters ?? {};
 
+    // to-do: get all events
     const events = await promiseMap(organisationUnits, async orgUnit => {
-        const { events } = await api.events
-            .getAll({
+        const { instances } = await api
+            .get<{ instances: { event: Id }[] }>("/tracker/events", {
                 program: program.id,
                 programStage: programStage?.id,
                 orgUnit: orgUnit.id,
-                startDate: startDate ? moment(startDate).format("YYYY-MM-DD") : undefined,
-                endDate: endDate ? moment(endDate).format("YYYY-MM-DD") : undefined,
+                occuredAfter: startDate ? moment(startDate).format("YYYY-MM-DD") : undefined,
+                occuredBefore: endDate ? moment(endDate).format("YYYY-MM-DD") : undefined,
+                fields: "event",
+                paging: false,
```

Comment:
I know normally we don't have thousands of events in a TEI and also we're filtering by different values (ou, ps, dates, etc), but using `paging=false` could lead to a 5XX response if the request include a lot of events.

Only if we have time it could be good to get 100-200 events by page and then recursive call the same method until the end of the pagination. You can find an example in the [user-extended repository](https://github.com/EyeSeeTea/user-extended-app/blob/development/src/data/repositories/UserD2ApiRepository.ts#L210)